### PR TITLE
Use custom exceptions instead of assertion errors

### PR DESCRIPTION
**Pull request by mcara**
Replaces `ValueError`  with a custom exception in the `Graph._sanity_check()` method.

### CODE REVIEW

File: spherical_geometry/graph.py [hunk 1]
```diff
@@ -375,7 +376,7 @@ def _sanity_check(self, title, node_is_2=False):
         for edge in self._edges:
             for node in edge._nodes:
                 if edge not in node._edges or node not in self._nodes:
-                    raise ValueError
+                    raise MalformedPolygonError(msg)
```
Comment (pllim) [inline spherical_geometry/graph.py:379]: Shouldn't sanity check raise `SanityError`? 😉

Comment (mcara) [reply spherical_geometry/graph.py:379]: Do you mean `Insanity`?

Comment (pllim) [reply spherical_geometry/graph.py:379]: Even better! 👍

Comment (mcara) [reply spherical_geometry/graph.py:379]: @pllim Could you please shine some light as to why the tests are being _`Expected`_ for over 6 hours?

**Comment by pllim** [comment]
Yes. It is because when you click on Actions tab, it tells you, "This scheduled workflow is disabled because there hasn't been activity in this repository for at least 60 days."

I have just re-enabled it for you. Another push would run the CI.

**Comment by mcara** [comment]
I have tested this locally and it worked.

**Pull request merged**